src/main/java/htsjdk/variant/variantcontext/VariantContextComparator.java

```diff
@@ -83,20 +97,24 @@ public int compare(final VariantContext firstVariantContext, final VariantContex
 		// Will throw NullPointerException -- happily -- if either of the chromosomes/contigs aren't
 		// present. This error checking should already have been done in the constructor but it's left
 		// in as defence anyway.
-		int contigCompare = this.contigIndexLookup.get(firstVariantContext.getContig()).compareTo(this.contigIndexLookup.get(secondVariantContext.getContig()));
-		contigCompare = contigCompare == 0 ? firstVariantContext.getStart() - secondVariantContext.getStart() : contigCompare;
-		if (contigCompare == 0) {
-			// Compare variants that have the same genomic span (chr:start-end) lexicographically by all alleles (ref and alts).
-			for (int i = 0; i < firstVariantContext.getAlleles().size(); i++) {
-				// If all previous alleles are identical and the first variant has additional alleles, make the first variant greater.
-				if (i >= secondVariantContext.getAlleles().size()) { return 1; }
-				contigCompare = firstVariantContext.getAlleles().get(i).compareTo(secondVariantContext.getAlleles().get(i));
-				if (contigCompare != 0) return contigCompare;
+		return internalComparator.compare(firstVariantContext, secondVariantContext);
+	}
+
+	//Sorts and uniques the inputs and then compares the values pairwise in their natural order
+	//until there is a mismatch or 1 of the inputs runs out.
+	private static class SortedUniquedElementsComparator<T extends Comparable<T>> implements Comparator<Collection<T>>{
+		@Override
+		public int compare(final Collection<T> o1, final Collection<T> o2) {
+			final Iterator<T> left = new TreeSet<>(o1).iterator();
```

Comment:
Perhaps is worth to add trivial cases to avoid TreeSet constructons since quite often alt-alleles list are just 1 in length?